### PR TITLE
fix: Update the parent consistently

### PR DESCRIPTION
**Pull request by draftcode**
When a conflict happens, in some cases the parent is updated before
conflict is resolved. When sync is aborted, the parent is already moved,
so it won't get back to the original state. The parent update should be
done after --continue.

Also, the parent should be updated even if it's not trunk. It should
update the parent hash as well.

This change creates a sync continuation for all cases. The continuation
contains the data needed for updating parents. The continuation is
called inline when there's no conflict. If there's a conflict, the
continuation is saved and applied later.

### CODE REVIEW

File: internal/actions/sync_branch.go [hunk 1]
```diff
@@ -435,29 +403,29 @@ func syncBranchContinue(
...
 	return &SyncBranchResult{*rebase, nil, branch}, nil
 }
 
-func syncBranchUpdateNewTrunk(
+func syncBranchUpdateParent(
 	tx meta.WriteTx,
 	branch meta.Branch,
-	newTrunk string,
+	continuation *SyncBranchContinuation,
 ) (meta.Branch, error) {
```
Comment (draftcode) [inline internal/actions/sync_branch.go:418]: Actually, this has a bug. I'll fix that. We should addremove the child/parents properly.

Comment (draftcode) [reply internal/actions/sync_branch.go]: Fixed. Now this adds a child when the parent is changed and the parent is not trunk.

File: internal/actions/sync_branch.go [hunk 2]
```diff
@@ -43,14 +42,10 @@ type SyncBranchResult struct {
 
 type (
 	SyncBranchContinuation struct {
```
Comment (travigd) [inline internal/actions/sync_branch.go:44]: Maybe makes sense to just rename this to `SyncBranchState` at this point?

**Review by travigd — APPROVED** [review]

File: internal/actions/sync_branch.go (hunk 2 above)
Comment (draftcode) [reply internal/actions/sync_branch.go:44]: I think continuation makes sense. It's indeed the continuation post rebase.

**Pull request merged**